### PR TITLE
continue gs backend

### PR DESCRIPTION
**Pull request by johmathe**

### CODE REVIEW

File: geomstats/backend/numpy_linalg.py [hunk 1]
```diff
@@ -5,3 +5,15 @@
...
 def norm(*args, **kwargs):
     return np.linalg.norm(*args, **kwargs)
+
+
+def inv(*args, **kwargs):
+    return np.linalg.inv(*args, **kwargs)
+
+
+def matrix_rank(*args, **kwargs):
+    return np.linalg.matrix_rank(*args, **kwargs)
```
[the remark this answers is not in the document]
Comment (johmathe) [reply geomstats/backend/numpy_linalg.py]: no - this is actually part of the backend

File: geomstats/general_linear_group.py [hunk 2]
```diff
@@ -3,10 +3,10 @@
 i.e. the matrix group GL(n).
 """
 
-import numpy as np
 import scipy.linalg
 
 from geomstats.lie_group import LieGroup
+import geomstats.backend as gs
 import geomstats.vectorization as vectorization
```
[the remark this answers is not in the document]
Comment (johmathe) [reply geomstats/general_linear_group.py]: good catch

File: geomstats/general_linear_group.py [hunk 3]
```diff
@@ -36,23 +36,23 @@ def belongs(self, mat):
         mat = vectorization.to_ndarray(mat, to_ndim=3)
```
Comment (ninamiolane) [inline geomstats/general_linear_group.py]: gs.to_ndarray?

File: geomstats/general_linear_group.py [hunk 4]
```diff
@@ -62,7 +62,7 @@ def group_exp_from_identity(self, tangent_vec):
         if tangent_vec.ndim == 2:
             return scipy.linalg.expm(tangent_vec)
 
-        exp = np.zeros_like(tangent_vec)
+        exp = gs.zeros_like(tangent_vec)
         n_vecs, _, _ = tangent_vec.shape
         for i in range(n_vecs):
             exp[i] = scipy.linalg.expm(tangent_vec[i])
```
Comment (ninamiolane) [inline geomstats/general_linear_group.py]: I will replace scipy's expm by gs' expm in a next pr.

File: geomstats/hyperbolic_space.py [hunk 5]
```diff
@@ -191,19 +188,19 @@ def exp_basis(self, tangent_vec, base_point):
...
         projected_tangent_vec = self.projection_to_tangent_space(
             vector=tangent_vec, base_point=base_point)
-        diff = np.abs(projected_tangent_vec - tangent_vec)
-        if not np.allclose(diff, 0):
+        diff = gs.abs(projected_tangent_vec - tangent_vec)
+        if not gs.allclose(diff, 0):
             tangent_vec = projected_tangent_vec
             logging.warning(
-                'The input vector is not tangent to the hyperbolic space.'
+                'The igsut vector is not tangent to the hyperbolic space.'
```
Comment (ninamiolane) [inline geomstats/hyperbolic_space.py]: ! input :)

Comment (johmathe) [reply geomstats/hyperbolic_space.py]: LOL

**Pull request merged**